File: solutions/feb_2024/2971_find_polygon_with_the_largest_perimeter/solution.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class Solution {
  public long largestPerimeter(int[] nums) {
    Arrays.sort(nums);

    List<Long> prefixSum = new ArrayList<>();
    long curSum = 0;
    for (int num : nums) {
      curSum += num;
      prefixSum.add(curSum);
    }

    for (int i = nums.length - 1; i >= 1; i--) {
      if (prefixSum.get(i) < prefixSum.get(i - 1) * 2) {
        return prefixSum.get(i);
      }
    }

    return -1;
  }
}

File: solutions/feb_2024/2971_find_polygon_with_the_largest_perimeter/solution.py
from typing import List


class Solution:
    def largestPerimeter(self, nums: List[int]) -> int:
        nums.sort()

        cur_sum = 0
        prefix_sum = []
        for num in nums:
            cur_sum += num
            prefix_sum.append(cur_sum)

        for i in range(len(nums) - 1, 1, -1):
            if prefix_sum[i] < prefix_sum[i - 1] * 2:
                return prefix_sum[i]

        return -1
